### ISSUE TITLE
Minimum number of iterations?

### ISSUE THREAD

**Issue by ianhays** [Microsoft/xunit-performance#91]
Think something like this may have value? I've noticed from my data that many tests will only run one or two iterations and the data will be skewed oddly. An example:

```
      <test name="System.Diagnostics.Tests.Perf_Process.Kill" type="System.Diagnostics.Tests.Perf_Process" method="Kill" time="11.4533988" result="Pass">
        <performance runid="System.Diagnostics.Process.Tests.dll-WindowsCore" etl="D:\git\corefx\bin\tests\Windows_NT.AnyCPU.Release\System.Diagnostics.Process.Tests\dnxcore50\System.Diagnostics.Process.Tests.dll-WindowsCore.etl">
          <metrics>
            <Duration displayName="Duration" unit="msec" />
          </metrics>
          <iterations>
            <iteration index="0" Duration="248.70042802554235" />
            <iteration index="1" Duration="93.757350318643148" />
          </iterations>
        </performance>
      </test>
```

It seems like we would want to run this test a few more times to get a better idea of the expected duration.

**Comment by ianhays** [comment]
Related to #8

**Comment by pharring** [comment]
I agree. There are some tests (e.g. Roslyn "solution build" tests) where I want to say, for example: Always run 5 iterations; no more, no less. The minimum is specified because there might be machine-wide caching behaviors the are not detectable in the xunit-performance framework itself. The upper bound is specified because I want to keep a lid on total execution time (e.g. my runs start at midnight and I want results before 9am)

**Comment by pharring** [comment]
The implementation uses an environment variable which affects all test methods equally in a run.
I expected to see parameters on the Benchmark attribute so that each test method could make that determination.

**Comment by brianrob** [comment]
Closing as I believe this is addressed by #142.
